### PR TITLE
Add conceptual docs on service registry and registrar

### PR DESCRIPTION
**Pull request by davidkline-ms**
This document strives to answer the questions:

- What are the MixedRealityServiceRegistry and IMixedRealityServiceRegistrar?
- Why two?
- How are they related?

### CODE REVIEW

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# What are the MixedRealityServiceRegistry and IMixedRealityServiceRegistrar
+
+The Mixed Reality Toolkit has two very similarly named components that perform related tasks: 
+MixedRealityServiceRegistry and IMixedRealityServiceRegistrar. What are they and what purpose does each serve?
```
Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: use [`` MixedRealityServiceRegistry`](xref:blah blah)? and for registrar

Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: Remove "What are they and what purpose do they serve" redundant with the title

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# What are the MixedRealityServiceRegistry and IMixedRealityServiceRegistrar
```
Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: Add question mark at end? or is that against the doc guidelines?

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+if (!MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem))
+{
+    // Failed to acquire the input system. It may not have been registered
+}
+```
+
+## IMixedRealityServiceRegistrar
+
+The IMixedRealityServiceRegistrar is the interface that defines the interface implemented by
```
Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: reword: "interface that defines the interface"

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# What are the MixedRealityServiceRegistry and IMixedRealityServiceRegistrar
+
+The Mixed Reality Toolkit has two very similarly named components that perform related tasks: 
+MixedRealityServiceRegistry and IMixedRealityServiceRegistrar. What are they and what purpose does each serve?
+
+## MixedRealityServiceRegistry
+
+The MixedRealityServiceRegistry is the component that contains instances of each registered service
+(core systems and extension services). It MixedRealiityServiceRegistry is implemented as a static C#
```
Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: either here or maybe else where call out that a service is defined by a IMixedRealityService?

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+components that manage the registration of one or more services. Components that implement
+IMixedRealityServiceRegistrar are responsible for adding and removing the data within the 
+MixedRealityServiceRegistry. The MixedRealityToolkit object is
+one such component.
+
+Other registrars can be cound in the MixedRealityToolkit.SDK.Experimental.Features
+folder. These components can be used to add single servce (ex: Spatial Awareness) support
+to an application. These single service managers are listed below.
+
+- BoundarySystemManager
```
Comment (Troy-Ferrell) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: xref as well? Or do we not xref experimental namespace?

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 4 above)
Comment (wiwei) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: >MixedRealiityServiceRegistry [](start = 42, length = 28)

Nit: spelling

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+- CameraSystemManager
+- DiagnosticsSystemManager
+- InputSystemManager
+- SpatialAwarenessSysstemManager
```
Comment (wiwei) [inline Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: >SpatialAwarenessSysstemManager [](start = 2, length = 30)

Nit: spelling

File: Documentation/toc.yml [hunk 7]
```diff
@@ -128,7 +128,11 @@
     - name: Using the Visual Profiler
       href: Diagnostics/UsingVisualProfiler.md
   - name: Services
-    href: MixedRealityServices.md
+    items:
+    - name: What makes a mixed reality feature
+      href: MixedRealityServices.md
+    - name: What are the MixedRealityServiceRegistry and IMixedRealityServiceRegistrar
+      href: ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md
```
Comment (wiwei) [inline Documentation/toc.yml]: Consider moving this content to the Architecture section above, as this does feel like Architecture-y sort of stuff (i.e. it explains system creation/setup/destruction stuff)

**Comment by davidkline-ms** [comment]
Great feedback @Troy-Ferrell! I'll make the changes

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 1 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: added

Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: removed

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 5 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: added

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 4 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: fixed

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 6 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: fixed

File: Documentation/toc.yml (hunk 7 above)
Comment (davidkline-ms) [reply Documentation/toc.yml]: i believe it is less discoverable there,

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 3 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: fixed

File: Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md (hunk 4 above)
Comment (davidkline-ms) [reply Documentation/ServiceUtilities/MixedRealityServiceRegistryAndIMixedRealityServiceRegistrar.md]: added

**Pull request merged**